default.nix

```diff
@@ -1,16 +1,98 @@
-with (import (fetchTarball https://github.com/NixOS/nixpkgs/archive/ffea68a09cedab941d19f02c9041689ebc81192e.tar.gz) {});
-# NOTE: when bumping nixpkgs, also update nixpkgs-src.json and .travis.yml
+let
+  defaults = {
+    master_config = {
+      # TODO DEVOPS-673
+      cardano_rev = "eef095";
+      cardano_hash = "151qjqswrcscr2afsc6am4figw09hyxr80nd3dv3c35dvp2xx4rp";
+    };
+    pkgs = import (import ./fetchNixpkgs.nix (builtins.fromJSON (builtins.readFile ./nixpkgs-src.json))) { config = {}; overlays = []; };
+  };
+in { cluster ? "mainnet", master_config ? defaults.master_config, pkgs ? defaults.pkgs, buildNr ? "nix" }:
```

Comment:
btw, top-level function arguments are recursive you can include defaults here:

```
{ cluster ? "mainnet"
, master_config ? { ... }
, pkgs ? import (import ./fetchNixpkgs.nix (builtins.fromJSON (builtins.readFile ./nixpkgs-src.json))) { config = {}; overlays = []; }
, buildNr ? "nix" }:
```

Reply:
yeah, but that gets a bit messy with the nixpkgs stuff in there
the master_config is also going away with the changes to pin cardano